packages/manager/modules/pci/src/projects/project/storages/cold-archive/add/components/steps/link-user-archive/controller.js

```diff
@@ -247,14 +251,22 @@ export default class ColdArchiveLinkUserArchiveController {
         )
       : Service.generateS3Credentials(this.projectId, user.id);
 
+    const secretPromise = this.PciStoragesObjectStorageService.getS3Secret(
+      this.projectId,
+      user.id,
+      user.s3Credentials?.access,
+    );
     this.userModel.linkedMode.isInProgress = true;
-    return taskPromise
-      .then((credential) => {
+    return Promise.all([taskPromise, secretPromise])
```

Comment:
Prefer using `$q` instead
```suggestion
    return this.$q.all([taskPromise, secretPromise])
```